### PR TITLE
Added types to queries to persistence.

### PR DESCRIPTION
**Pull request by julien-sebire**
This is caused by https://github.com/oat-sa/generis/pull/684 interface change.
As it is adding optional argument to method, it does not break interfaces and can be merged directly.
To test please update config/generis/persistences.conf.php
```
        'default' => array(
            'driver' => \oat\taoDevTools\models\persistence\SqlProxyDriver::class,
            'persistenceId' => 'inner'
        ),
        'inner' => array(
            'driver' => 'dbal',
            'connection' => array(
                'driver' => 'pdo_mysql',
                'host' => 'localhost',
                'dbname' => 'tao_tao',
                'user' => 'root',
                'password' => 'r00t',
                'charset' => 'utf8'
            )
        ),```

### CODE REVIEW

File: models/persistence/SqlProxyDriver.php [hunk 1]
```diff
@@ -59,17 +59,18 @@ function connect($id, array $params)
...
-     * @param $params
+     * @param string $statement
+     * @param array $params
+     * @param array $types
      *
      * @return mixed
      * @throws DBALException
      */
-    public function query($statement, $params) 
+    public function query($statement, $params, array $types = [])
```
Comment (mike-ionut-mihai-sandu-tao) [inline models/persistence/SqlProxyDriver.php]: I think the contract `common_persistence_sql_Driver` 
has the signature : ` public function query($statement,$params);`
and i don't think changing the contract is the best solution

File: models/persistence/SqlProxyDriver.php [hunk 2]
```diff
@@ -78,17 +79,18 @@ public function query($statement, $params)
...
-     * @param $params
+     * @param string $statement
+     * @param array $params
+     * @param array $types
      *
      * @return mixed
      * @throws DBALException
      */
-    public function exec($statement, $params) 
+    public function exec($statement, $params, array $types = [])
```
Comment (mike-ionut-mihai-sandu-tao) [inline models/persistence/SqlProxyDriver.php]: same here

File: models/persistence/SqlProxyDriver.php (hunk 1 above)
Comment (julien-sebire) [reply models/persistence/SqlProxyDriver.php]: This is needed by the contact changed in the linked PR. It fixes the problems we have for example in the task queue. We need this to be able to provide field types for DBAL to perform correct value conversions for booleans in PostgreSql and for null values in Spanner.

Comment (siwane) [reply models/persistence/SqlProxyDriver.php]: This is a necessary in order to integrate generis signature changed. As optional parameter, it does not break the implementation

**Pull request merged**